pkg/client/dtclient/settings_client.go

```diff
@@ -247,6 +257,76 @@ func (d *DynatraceClient) upsertSettings(ctx context.Context, obj SettingsObject
 	return entity, nil
 }
 
+func isSameValueOfConstraint(key string, c1 []byte, c2 []byte) (bool, error) {
+	u := make(map[string]any)
+	if err := json.Unmarshal(c1, &u); err != nil {
+		return false, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+	}
+	v1 := u[key]
+
+	if err := json.Unmarshal(c2, &u); err != nil {
+		return false, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+	}
+	v2 := u[key]
+
+	return reflect.DeepEqual(v1, v2), nil
+}
+
+func (d *DynatraceClient) findObjectWithMatchingConstraints(ctx context.Context, source SettingsObject) (*DownloadSettingsObject, error) {
+	constraints, err := d.fetchSchemasConstraints(ctx, source.SchemaId)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get details for %q schema: %w", source.SchemaId, err)
+	}
+
+	if len(constraints.UniqueProperties) == 0 {
+		return nil, nil
+	}
+
+	objects, err := d.listSettings(ctx, source.SchemaId, ListSettingsOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get existing settings objects for %q schema: %w", source.SchemaId, err)
+	}
+
+	target, err := findObjectWithSameConstraints(constraints, source, objects)
+	if err != nil {
+		return nil, err
+	}
+	return target, nil
+}
+
+func findObjectWithSameConstraints(schema SchemaConstraints, source SettingsObject, objects []DownloadSettingsObject) (*DownloadSettingsObject, error) {
+	var candidate *DownloadSettingsObject
+	var err error
+	for _, rule := range schema.UniqueProperties {
+		for j, o := range objects {
+			var b bool
+			b, err = areObjectsEqualForTheRule(rule, source, o)
+			if b {
+				if candidate == nil {
+					candidate = &objects[j]
+				}
+				if candidate != &objects[j] {
+					err = errors.Join(err, fmt.Errorf("can't update or create new configuration %q couse the already existing objects with ID %q and %q interfearing", source.Coordinate, candidate.ObjectId, objects[j].ObjectId))
+				}
+			}
+		}
+	}
+
+	return candidate, err
+}
+
+func areObjectsEqualForTheRule(rule []string, source SettingsObject, object DownloadSettingsObject) (bool, error) {
```

Comment:
I think the code that handles this is hard for me to wrap my head around. You need to know how `isSameValueOfConstraint` works internally to understand that the error is handled correctly.

I would go with a less clever but more straightforward solution: Checking the error explicitly, and also not returning the variable `b`, as it is clever, but the constants true/false are clearer.

I would suggest the change as follows:
```
func areObjectsEqualForTheRule(rule []string, source SettingsObject, object DownloadSettingsObject) (bool, error) {
	var eq bool
	var err error
	for _, c := range rule {
		eq, err = isSameValueOfConstraint(c, object.Value, source.Content)
		if err != nil {
			return false, err
		}

		if !eq {
			return false, nil
		}
	}

	return true, err
}
```

Reply:
I've refactored this a bit in the last two commits I've added - I think it's easier to read now. please take another look